File: clase1-intro-nodejs/main.java
public class Persona {
    private String nombre;
    private int edad;
    private static int cantPersonas;

    public Persona(String nombre, int edad) {
        this.nombre = nombre;
        this.edad = edad;
        Persona.cantPersonas++;
    }

    public String getNombre() {
        return this.nombre;
    }

    public void setNombre(value) {
        if (value == null) {
            throw new IllegalArgumentException("el nombre es un campo requerido");
        }
        this.nombre = value;
    }

    public void saludar() {
        String saludo = String.format("hola! soy %s", this.nombre);
        System.out.print(saludo);
    }
}

public class Main {
    public static void main(String[] args) {
        String saludo = "hola mundo";
        System.out.println(saludo);

        Persona p = new Persona("marian", 36);
        System.out.println(p.nombre);

        p.setNombre("pepe");

        p.saludar();
    }
}

File: clase1-intro-nodejs/main.js
class MissingArgumentError extends Error {
    constructor(fieldName) {
        super("'" + fieldName + "' es un campo requerido")
    }
}

class Persona {
    #nombre
    #edad
    static #cantPersonas = 0

    constructor(nombre, edad) {
        this.#nombre = nombre
        this.#edad = edad
        Persona.#cantPersonas++
    }

    get nombre() {
        return this.#nombre
    }

    set nombre(value) {
        if (value == null) {
            throw new MissingArgumentError("nombre")
        }
        this.#nombre = value
    }

    saludar() {
        const saludo = `hola! soy ${this.nombre}, ya somos ${Persona.#cantPersonas}`
        console.log(saludo);
    }
}

const saludo = "hola mundo"
console.log(saludo)

const p = new Persona("marian", 36)
p.nombre = "pepe"
// p.nombre = null
console.log(p.nombre)

const p2 = new Persona("german", 20)

p2.saludar()